src/GraphQL/Extensions/AssemblyExtensions.cs

```diff
@@ -72,5 +72,14 @@ public static List<(Type ClrType, Type GraphType)> GetClrTypeMappings(this Assem
             //return the list of type mappings
             return typeMappings;
         }
+
+        internal static string? GetNuGetVersion(this Assembly assembly)
+        {
+            var version = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+            if (version == null)
+                return null;
+            var i = version.LastIndexOf('+');
```

Comment:
Are you OK with versions like `1.0.0-alpha+001`?

Reply:
No; I intended it to return the nuget package version such as `1.0.0` or `1.0.0-alpha`.  This is why this code exists.

Reply:
I mean versions like `1.0.0-alpha` after trimming metadata tail. Personally I'm fine either way. Why do you want to avoid metadata? Then why use `InformationalVersion` at all? Just use "normal" version x.y.x, no?

Reply:
I'm not sure what your confusion is here.  This method, called `GetNuGetVersion`, returns the NuGet version of the package, and does exactly that.  If it's NuGet version 3.0.0, then it returns "3.0.0".  If it's 4.0.5-preview1 then it returns "4.0.5-preview1".  Returning "3.0.0+952" or "4.0.5-preview1+153" is not the NuGet version number.

Reply:
Also, returning "4.0.5" for a NuGet package versioned 4.0.5-preview1 is also incorrect within the context of this method, as that is not the NuGet version number.

Reply:
ok